src/test/boost.simd.cpp

```diff
@@ -0,0 +1,49 @@
+
+//          Copyright John McFarlane 2015 - 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//  (See accompanying file ../../LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+#include <sg14/auxiliary/boost.simd.h>
+
+#include <sg14/fixed_point>
+
+#include <boost/simd/function/all.hpp>
+
+#include <gtest/gtest.h>
+
+namespace {
+    using sg14::fixed_point;
+    using boost::simd::pack;
+
+    template<class LhsT, std::size_t LhsN, int LhsExponent, class RhsT, std::size_t RhsN, int RhsExponent>
+    void assert_identical(const fixed_point <pack<LhsT, LhsN>, LhsExponent> &lhs,
+                          const fixed_point <pack<RhsT, RhsN>, RhsExponent> &rhs) {
+        static_assert(LhsN == RhsN, "pack is the wrong size");
+        static_assert(std::is_same<LhsT, RhsT>::value, "pack is the wrong type");
+        auto equal = lhs == rhs;
+        if (!boost::simd::all(equal)) {
```

Comment:
Will change this. Although in the same spirit of using `explicit` on conversion operators, have you considered an `explicit operator bool()`?